blocks/extensions/__init__.py

```diff
@@ -363,7 +363,7 @@ def __init__(self, **kwargs):
         super(FinishAfter, self).__init__(**kwargs)
 
     def do(self, which_callback, *args):
-        self.main_loop.log.current_row.training_finish_requested = True
+        self.main_loop.log.current_row['training_finish_requested'] = True
```

Comment:
I believe we've had this discussion before, but I think that would be a bit overly defensive. It introduces unnecessary new variables which need to be imported if you want to use them in other files (which people will probably just forget anyway), so you end up with a complicated web of import statements if you don't want to start redefining constants in each file (which would defeat the purpose). You end up with a whole bunch of them, which I think is ugly anyway, but it also seems arbitrary to use constants in some places and free strings in other. If you write to the wrong key, you'll now get a `KeyError` when you try to read it (instead of the `None` you used to get), so a typo should show itself quite quickly. In the end, catching typos is just one of the reasons that we do code reviews. For callbacks I think a custom `__eq__` was reasonable, because we're comparing against a limited, defined set of callbacks, but in this case I think we don't need to be as paranoid.